syft/frameworks/torch/tensors/interpreters/native.py

```diff
@@ -909,6 +909,11 @@ def share(
             requires_grad (bool): Should we add AutogradTensor to allow gradient computation,
                 default is False.
         """
+        if protocol == "falcon":
+            shared_tensor = syft.ReplicatedSharingTensor(owner=self.owner).share_secret(
```

Comment:
Q: I am thinking if we can keep a ```dict``` for those.
Like:
```dict = {
  "falcon": syft.ReplicatedSharingTensor,
  "snn"    : syft.AdditiveSharingTensor
}
```

Reply:
what would be the use-case of that dict? I don't get it

Reply:
Dict are better to dynamically add options if needed, and are faster to access instead of if / else-ing on all optiosn

Reply:
@abogaziah are you making the changes in this PR or we can create a subsequent ticket to do the changes in another issue (we can also mark it as good first issue :D)

Reply:
I think it needs to be done later, because sharing doesn't work the same way in both cases for now, (AST is more complex) maybe we should do that after refactoring AST

Reply:
Sounds good! There is also some refacto going on in the AST file in #3924 :)